packages/platforms/delta/x86-64/x86-64-delta-ak7448/onlp/builds/src/module/src/thermali.c

```diff
@@ -0,0 +1,165 @@
+            ONLP_THERMAL_CAPS_ALL, 0, ONLP_THERMAL_THRESHOLD_INIT_DEFAULTS
+        },
+    { { ONLP_THERMAL_ID_CREATE(THERMAL_2_ON_MAIN_BOARD), "Thermal sensor near MAC (U10)", 0},
+        ONLP_THERMAL_STATUS_PRESENT,
+        ONLP_THERMAL_CAPS_ALL, 0, ONLP_THERMAL_THRESHOLD_INIT_DEFAULTS
+        },
+    { { ONLP_THERMAL_ID_CREATE(THERMAL_3_ON_MAIN_BOARD), "Thermal sensor near middle SFP cage (U11)", 0},
+            ONLP_THERMAL_STATUS_PRESENT,
+            ONLP_THERMAL_CAPS_ALL, 0, dni_onlp_thermal_threshold(65000,75000,80000)
+        },
+    { { ONLP_THERMAL_ID_CREATE(THERMAL_4_ON_MAIN_BOARD), "Thermal sensor near QSFP cage (U13)", 0},
+            ONLP_THERMAL_STATUS_PRESENT,
+            ONLP_THERMAL_CAPS_ALL, 0, ONLP_THERMAL_THRESHOLD_INIT_DEFAULTS
+        },
+    { { ONLP_THERMAL_ID_CREATE(THERMAL_5_ON_FAN_BOARD), "Thermal sensor near FAN (U334)", 0},
+            ONLP_THERMAL_STATUS_PRESENT,
+            ONLP_THERMAL_CAPS_ALL, 0, ONLP_THERMAL_THRESHOLD_INIT_DEFAULTS
+        },
+    { { ONLP_THERMAL_ID_CREATE(THERMAL_1_ON_PSU1), "PSU-1 Thermal Sensor 1", ONLP_PSU_ID_CREATE(PSU1_ID)},
+            ONLP_THERMAL_STATUS_PRESENT,
+            ONLP_THERMAL_CAPS_ALL, 0, ONLP_THERMAL_THRESHOLD_INIT_DEFAULTS
+        },
+};
+
+/*
+ * This will be called to intiialize the thermali subsystem.
+ */
+int
+onlp_thermali_init(void)
+{
+    return ONLP_STATUS_OK;
+}
+
+/*
+ * Retrieve the information structure for the given thermal OID.
+ *
+ * If the OID is invalid, return ONLP_E_STATUS_INVALID.
+ * If an unexpected error occurs, return ONLP_E_STATUS_INTERNAL.
+ * Otherwise, return ONLP_STATUS_OK with the OID's information.
+ *
+ * Note -- it is expected that you fill out the information
+ * structure even if the sensor described by the OID is not present.
+ */
+int
+onlp_thermali_info_get(onlp_oid_t id, onlp_thermal_info_t* info)
+{
+    int temp_base = 1;
+    int local_id, r_data;
+    char  fullpath[256] = {0};
+    uint8_t channel, offset;
+    mux_info_t mux_info;
+
+    VALIDATE(id);
+    local_id = ONLP_OID_ID_GET(id);
+
+    /* Set the onlp_oid_hdr_t and capabilities */
+    *info = linfo[local_id];
+
+    if(local_id == THERMAL_CPU_CORE) {
+        int rv = onlp_file_read_int_max(&info->mcelsius, cpu_coretemp_files);
+        return rv;
+    }
+
+    switch (local_id) {
+        case THERMAL_5_ON_FAN_BOARD:
+            offset = FAN_I2C_MUX_SEL_REG;
+            channel = FAN_I2C_SEL_THERMAL;
+            break;
+        case THERMAL_1_ON_PSU1:
+            offset = PSU_I2C_MUX_SEL_REG;
+            channel = PSU_I2C_SEL_PSU_EEPROM;
+            break;
+        default:
+            channel = 0x00; /* DEFAULT */
+            break;
+    }
+
+    mux_info.bus = I2C_BUS_5;
+    mux_info.addr = CPLD_B;
+    mux_info.offset = offset;
+    mux_info.channel = channel;
+    mux_info.flags = DEFAULT_FLAG;
+    
+    sprintf(fullpath, "%s%s", PREFIX_PATH, last_path[local_id]);
+    r_data = dni_i2c_lock_read_attribute(&mux_info, fullpath);
```

Comment:
Call onlp_thermali_info_get() to read  the attribute by dni_i2c_lock_read_attribute() function.